pytorch_lightning/core/lightning.py

```diff
@@ -373,14 +381,27 @@ def all_gather(self, tensor: Union[torch.Tensor], group: Optional[Any] = None, s
         distributed processes
 
         Args:
-            tensor: tensor of shape (batch, ...)
+            tensor: int, float, tensor of shape (batch, ...),  or a collection of
+                int, float, tensor of shape (batch, ...)
             group: the process group to gather results from. Defaults to all processes (world)
             sync_grads: flag that allows users to synchronize gradients for all_gather op
 
         Return:
-            A tensor of shape (world_size, batch, ...)
+            A tensor of shape (world_size, batch, ...), or if the input was a collection
+            the output will also be a collection with tensors of this shape.
         """
-        return self.trainer.accelerator_backend.all_gather(tensor, group=group, sync_grads=sync_grads)
+        if self.trainer.accelerator_backend is not None:
+            all_gather = self.trainer.accelerator_backend.all_gather
+        else:
+            all_gather = all_gather_ddp_if_available
+
+        def to_dtype_tensor(value, dtype=None):
+            return torch.tensor(value, dtype=dtype, device=self.device)
+
+        data = apply_to_collection(data, float, partial(to_dtype_tensor, dtype=torch.float))
+        data = apply_to_collection(data, int, partial(to_dtype_tensor, dtype=torch.int))
```

Comment:
Do you know if torch.long/half are supported?

Reply:
add support for bool and np.ndarray.